cmd/monaco/account/deploy.go

```diff
@@ -107,23 +114,39 @@ func deploy(fs afero.Fs, opts deployOpts) error {
 	log.Debug("Deploying to accounts: %q", maps.Keys(accs))
 	log.Debug("Deploying projects: %q", maps.Keys(projs))
 
-	resources, errs := loadAccountManagementResources(fs, projs)
+	resources, errs := loadAccountManagementResources(fs, opts.workingDir, projs)
 	if len(errs) > 0 {
 		errutils.PrintErrors(errs)
 		return errors.New("failed to load all account management resources")
 	}
 
-	err := deployer.Deploy(accs, resources, deployer.Options{DryRun: opts.dryRun})
-	return err
+	accountClients, err := createAccountClients(accs)
+	if err != nil {
+		return fmt.Errorf("failed to creae account clients: %w", err)
+	}
+
+	supportedPermissions, err := deployer.DefaultPermissionProvider()
```

Comment:
nit: it's a bit weird that the caller needs to know and explicitly call the default permissions. IMO might be nicer to do this as part of the deployer/client initalization. Assuming you're only overwriting them for testing anyway, maybe you can just initialize the client directly for tests?

But at a second glance I also realize you query them once, but build several clients.. In a real scenario I'd expect most customers to not really have more than one account, so maybe we could live with the inefficiency - or cache as part of a DefaultProvider that's not just a function

Reply:
Let possibly target this in another PR.